quixstreams/state/recovery.py

```diff
@@ -140,16 +168,35 @@ class ChangelogProducer:
     kafka changelog partition.
     """
 
-    def __init__(self, changelog_name: str, partition_num: int, producer: RowProducer):
+    def __init__(
+        self,
+        changelog_name: str,
+        source_topic_name: str,
+        partition: int,
+        producer: RowProducer,
+    ):
         """
         :param changelog_name: A changelog topic name
-        :param partition_num: source topic partition number
+        :param partition: source topic partition number
         :param producer: a RowProducer (not shared with `Application` instance)
         """
         self._changelog_name = changelog_name
-        self._partition_num = partition_num
+        self._source_topic_name = source_topic_name
+        self._partition = partition
```

Comment:
This is more just a general comment:

in case you were wondering at the extra verbosity around things like `partition` vs `partition_num`, wanted it to be more obvious at a glance whether the objects we were referring to were either `Topic` objects or `topic` names, or `Partition`-like objects (our RDB objects) or actual Kafka partition numbers.

In some ways it's a little redundant with everything typed already, but I found it helped me while reading the code sometimes since we kinda call these different objects using the same variable names within a similar, overlapping context. 

Feel free to keep your changes if you feel it's not useful, but I thought I'd expand a little in case you didn't know why I had it that way.

Reply:
The changes here were made to keep things consistent across the code. 
I suggest we use "partition" when we refer to the Kafka partition numbers, "topic_partition" or "tp" when it's a `TopicPartition` instance, and "state_partition" when it's the `StatePartition` instance.